packages/node-utils/src/tests/http.test.ts

```diff
@@ -0,0 +1,155 @@
+        const middlewareHandler = jest.fn((_request, response, next) => {
+            response.setHeader('foo', 'bar');
+            next();
+        });
+        const handler = jest.fn((_request, response) => {
+            response.end('ok');
+        });
+        server.get('/foo', [middlewareHandler, handler]);
+        await server.start();
+        const address = server.getServerAddress();
+        expect(address).toBeDefined();
+        const res = await fetch(`http://${address.address}:${address.port}/foo`);
+        expect(res.headers.get('foo')).toEqual('bar');
+    });
+
+    test('calling endpoint which as no defined handler', async () => {
+        await server.start();
+        const address = server.getServerAddress();
+        expect(address).toBeDefined();
+
+        fetch(`http://${address.address}:${address.port}/foo`).catch(_err => {});
+
+        await new Promise(resolve => setTimeout(resolve, 1000));
+
+        server.stop();
+
+        await new Promise<void>(resolve => {
+            server.on('server/closed', () => {
+                resolve();
+            });
+        });
+    });
+
+    test('aborting request', async () => {
+        const controller = new AbortController();
+        const { signal } = controller;
+        server.get('/foo', [
+            (_request, response) => {
+                setTimeout(() => response.end('ok'), 1000);
+            },
+        ]);
+        await server.start();
+
+        const address = server.getServerAddress();
+        expect(address).toBeDefined();
+        fetch(`http://${address.address}:${address.port}/foo`, { signal }).catch(_err => {});
+        await new Promise(resolve => setTimeout(resolve, 500));
+        controller.abort();
+        await new Promise(resolve => setTimeout(resolve, 500));
+        expect(muteLogger.info).toHaveBeenLastCalledWith(
+            expect.any(String),
+            'Request /foo aborted',
+        );
+    });
+
+    test('allowOrigin middleware lets request with allowed origin through', async () => {
+        const handler = jest.fn((_request, response) => {
+            response.end('ok');
+        });
+
+        server.get('/foo', [allowOrigins(['*']), allowOrigins(['*.meow.com']), handler]);
+
+        server.get('/foo-bar', [
+            // empty string is allowed origin when referer is not defined
+            allowOrigins(['']),
+            handler,
+        ]);
+        await server.start();
+
+        const address = server.getServerAddress();
+        expect(address).toBeDefined();
+        const res1 = await fetch(`http://${address.address}:${address.port}/foo`, {
+            headers: {
+                referer: 'http://sub.meow.com',
+            },
+        });
+
+        const res2 = await fetch(`http://${address.address}:${address.port}/foo-bar`);
+
+        expect(res1.status).toEqual(200);
+        expect(res2.status).toEqual(200);
+    });
```

Comment:
I've added a unit test. Should the handler order in this case be `use1` -> `use2` -> `get1`? Because it's not. I'll try to fix it.

```suggestion
    });

    test('handler order', async () => {
        const calledHandlers: string[] = [];
        const handler = (name: string, end?: boolean) => (_: any, res: any, next: () => void) => {
            calledHandlers.push(name);
            if (end) res.end('ok');
            else next();
        };

        server.use([handler('use1'), handler('use2')]);
        server.post('/foo', [handler('post1', true)]);
        server.get('/foo', [handler('get1', true)]);

        await server.start();
        const { address, port } = server.getServerAddress();

        await fetch(`http://${address}:${port}/foo`);

        expect(calledHandlers).toStrictEqual(['use1', 'use2', 'get1']);
    });
```